storm-core/src/jvm/backtype/storm/messaging/netty/Client.java

```diff
@@ -182,7 +177,7 @@ private boolean connectionEstablished(Channel channel) {
         // See:
         // - http://netty.io/3.9/api/org/jboss/netty/channel/ChannelEvent.html
         // - http://stackoverflow.com/questions/13356622/what-are-the-netty-channel-state-transitions
-        return channel != null && channel.isConnected();
+        return channel != null && channel.isOpen();
```

Comment:
Why the change from connected to open? the channel could be open, but not yet in the connected state, and this could cause a bug.

Unless Netty 4 has changed the model....